gadget/install/encrypt_test.go

```diff
@@ -58,78 +58,123 @@ func (s *encryptSuite) SetUpTest(c *C) {
 	c.Assert(os.MkdirAll(dirs.SnapRunDir, 0755), IsNil)
 }
 
-func (s *encryptSuite) TestEncryptHappy(c *C) {
-	s.mockCryptsetup = testutil.MockCommand(c, "cryptsetup", "")
-	s.AddCleanup(s.mockCryptsetup.Restore)
-
-	// create empty key to prevent blocking on lack of system entropy
-	key := secboot.EncryptionKey{}
-	dev, err := install.NewEncryptedDevice(&mockDeviceStructure, key, "some-label")
-	c.Assert(err, IsNil)
-	c.Assert(dev.Node, Equals, "/dev/mapper/some-label")
-
-	c.Assert(s.mockCryptsetup.Calls(), DeepEquals, [][]string{
+func (s *encryptSuite) TestNewEncryptedDevice(c *C) {
+	for _, tc := range []struct {
+		formatErr error
```

Comment:
Some small comments would be nice here, i.e. we have three times "err" - either some small comment or soemthing like:
```
mockedFormatErr
mockedOpenErr
expectedErr
```
(or similar)